pkgs/development/python-modules/pip-tools/default.nix

```diff
@@ -25,11 +26,14 @@ buildPythonPackage rec {
     hash = "sha256-Oeiu5GVEbgInjYDb69QyXR3YYzJI9DITxzol9Y59ilU=";
   };
 
+  patches = [ ./fix-setup-py-bad-syntax-detection.patch ];
```

Comment:
Can you add a comment with a link to the upstream issue? It helps future maintainers quickly find out the context and know if the patch is still applicable.

Reply:
I didn't open an upstream issue because I can't reproduce this behaviour outside of the build environment, if I run this locally:
```python
from build import BuildException
from build.util import project_wheel_metadata
import os

try:
    metadata = project_wheel_metadata(
        os.path.dirname(os.path.abspath("setup.py"))
    )
except StopIteration as e:
    print(f"exception: {e}")
```
I get:
`build.BuildException: Source /tmp/tmp.T1x1RXQnK6 does not appear to be a Python project: no pyproject.toml or setup.py`

I guess the `StopIteration` exception must be specific to pytest or the build env.

Reply:
It looks like the patch should be changed to catch both exceptions then.

Reply:
Fixed.